assets/js/blocks/checkout/inner-blocks/checkout-collection-method-block/edit.tsx

```diff
@@ -164,7 +164,21 @@ export const Edit = ( {
 	} );
 	const { showPrice, showIcon, className, localPickupText, shippingText } =
 		attributes;
-	const { shippingRates } = useShippingData();
+	const {
+		shippingRates,
+		needsShipping,
+		hasCalculatedShipping,
+		isCollectable,
+	} = useShippingData();
+
+	if (
+		! needsShipping ||
+		! hasCalculatedShipping ||
+		! shippingRates ||
+		! isCollectable
+	) {
+		return null;
+	}
```

Comment:
Why are we returning null on the edit function and not the block one, are we expecting different conditions?

Reply:
We return null here to avoid rendering the form step.